cmake/Modules/FindGio.cmake

```diff
@@ -0,0 +1,31 @@
+# - Try to find Gio
+# Once done this will define
+#
+#  GIO_FOUND - system has Gio
+#  GIO_INCLUDE_DIRS - the Gio include directory
+#  GIO_LIBRARIES - the libraries needed to use Gio
+#  GIO_DEFINITIONS - Compiler switches required for using Gio
+
+if(NOT WIN32)
+
+  # Use pkg-config to get the directories and then use these values
+  # in the find_path() and find_library() calls
+  find_package(PkgConfig)
+  pkg_check_modules(PC_GIO gio-2.0)
+
+  set(GIO_DEFINITIONS ${PC_GIO_CFLAGS})
+
+  find_path(GIO_INCLUDE_DIRS gio.h PATHS ${PC_GIO_INCLUDEDIR} ${PC_GIO_INCLUDE_DIRS} PATH_SUFFIXES glib-2.0/gio/)
+  find_library(GIO_LIBRARIES NAMES gio-2.0 libgio-2.0 PATHS ${PC_GIO_LIBDIR} ${PC_GIO_LIBRARY_DIRS})
+
+  if(GIO_INCLUDE_DIRS AND GIO_LIBRARIES)
```

Comment:
Use find_package_handle_standard_args if you are writing a new find module. Perhaps there is already one being maintained in upstream cmake already?

Reply:
Hm, what's "upstream" here? As far as I can see, GLib itself doesn't ship any CMake modules.

Reply:
In any case, switched to `find_package_handle_standard_args`